pages/_app.jsx

```diff
@@ -0,0 +1,22 @@
+import GlobalStyle from "../styles";
+import initialGames from "@/public/initialGames.js";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import { uid } from "uid";
+
+export default function App({ Component, pageProps }) {
+  const router = useRouter();
+  const [games, setGames] = useState(initialGames);
+
+  function handleCreateGame(newGame) {
+    setGames([...games, { id: uid(), progress: "Planned", ...newGame }]);
+    router.push("/");
```

Comment:
In your user story and its wireframe you mention that users will be redirected to the detail page of the newly created game. So instead of using router.push to redirect a user to the homepage, you should alter this to be the link to the newly created game.

```suggestion
    const id = uid();
    setGames([...games, { id, progress: "Planned", ...newGame }]);
    router.push(`/${id}`);
```